### DISCUSSION TITLE
TypeError with 3-Phase PDU

### DISCUSSION THREAD

**Discussion by iunderwood** [netbox-community/netbox#11003] [General]
Here's an odd one that i ran across today.

`<class 'TypeError'>

unsupported operand type(s) for /: 'str' and 'int'

Python version: 3.8.10
NetBox version: 3.3.8`

I am having a problem accessing one particular kind of RPDU that I have defined in my environment.  The error that NetBox throws is this:

The trace from the log is this:
`Internal Server Error: /dcim/devices/3244/
Traceback (most recent call last):
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 84, in view
    return self.dispatch(request, *args, **kwargs)
  File "/opt/netbox/netbox/utilities/views.py", line 90, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 119, in dispatch
    return handler(request, *args, **kwargs)
  File "/opt/netbox/netbox/netbox/views/generic/object_views.py", line 68, in get
    return render(request, self.get_template_name(), {
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/shortcuts.py", line 24, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/backends/django.py", line 62, in render
    return self.template.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 175, in render
    return self._render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
    return self.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 157, in render
    return compiled_parent._render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
    return self.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 157, in render
    return compiled_parent._render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
    return self.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 157, in render
    return compiled_parent._render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
    return self.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 157, in render
    return compiled_parent._render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
    return self.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 63, in render
    result = block.nodelist.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
    return self.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 63, in render
    result = block.nodelist.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
    return self.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 63, in render
    result = block.nodelist.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
    return self.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/defaulttags.py", line 322, in render
    return nodelist.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
    return self.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/defaulttags.py", line 238, in render
    nodelist.append(node.render_annotated(context))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
    return self.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/defaulttags.py", line 542, in render
    return self.nodelist.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
    return self.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/defaulttags.py", line 238, in render
    nodelist.append(node.render_annotated(context))
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated
    return self.render(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1059, in render
    output = self.filter_expression.resolve(context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 739, in resolve
    new_obj = func(obj, *arg_vals)
  File "/opt/netbox/netbox/utilities/templatetags/helpers.py", line 131, in divide
    return round(x / y)
TypeError: unsupported operand type(s) for /: 'str' and 'int'`

The problem was originally observed in 3.3.7, and for grins, I upgraded the development environment to 3.3.8.  If I create a new PDU with the same device type, I have no problem accessing it.

There were no problems in the upgrade to 3.3.7, or 3.3.4 and 3.3.1 before that.

Where should I look next or grab information from?

**Comment by mtinberg** [comment]
so what is setting "x" on this line and you work backward through the traceback to find out where that data is coming from?  "1" (the letter) and 1 (the number) are two different things, it's getting a string of letters where it expects a number and is trying to divide by it, figuring out what the value it's erroring on and where that's introduced will lead you to the solution, probably.

File "/opt/netbox/netbox/utilities/templatetags/helpers.py", line 131, in divide return round(x / y) TypeError: unsupported operand type(s) for /: 'str' and 'int'

so I would start working your way up the traceback and also just look at the record you were displaying using the API browser /api/ and see if anything looks off, esp around the tags data


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: Ian Underwood ***@***.***>
Sent: Tuesday, November 22, 2022 12:15 PM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] TypeError with one kind of PDU (Discussion #11003)


Here's an odd one that i ran across today.

`<class 'TypeError'>

unsupported operand type(s) for /: 'str' and 'int'

Python version: 3.8.10
NetBox version: 3.3.8`

I am having a problem accessing one particular kind of RPDU that I have defined in my environment. The error that NetBox throws is this:

The trace from the log is this:
Internal Server Error: /dcim/devices/3244/ Traceback (most recent call last): File "/opt/netbox/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner response = get_response(request) File "/opt/netbox/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response response = wrapped_callback(request, *callback_args, **callback_kwargs) File "/opt/netbox/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 84, in view return self.dispatch(request, *args, **kwargs) File "/opt/netbox/netbox/utilities/views.py", line 90, in dispatch return super().dispatch(request, *args, **kwargs) File "/opt/netbox/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 119, in dispatch return handler(request, *args, **kwargs) File "/opt/netbox/netbox/netbox/views/generic/object_views.py", line 68, in get return render(request, self.get_template_name(), { File "/opt/netbox/venv/lib/python3.8/site-packages/django/shortcuts.py", line 24, in render content = loader.render_to_string(template_name, context, request, using=using) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/loader.py", line 62, in render_to_string return template.render(context, request) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/backends/django.py", line 62, in render return self.template.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 175, in render return self._render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render return self.nodelist.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp> return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated return self.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 157, in render return compiled_parent._render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render return self.nodelist.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp> return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated return self.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 157, in render return compiled_parent._render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render return self.nodelist.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp> return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated return self.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 157, in render return compiled_parent._render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render return self.nodelist.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp> return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated return self.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 157, in render return compiled_parent._render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/test/utils.py", line 111, in instrumented_test_render return self.nodelist.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp> return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated return self.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 63, in render result = block.nodelist.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp> return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated return self.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 63, in render result = block.nodelist.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp> return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated return self.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/loader_tags.py", line 63, in render result = block.nodelist.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp> return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated return self.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/defaulttags.py", line 322, in render return nodelist.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp> return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated return self.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/defaulttags.py", line 238, in render nodelist.append(node.render_annotated(context)) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated return self.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/defaulttags.py", line 542, in render return self.nodelist.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in render return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1000, in <listcomp> return SafeString("".join([node.render_annotated(context) for node in self])) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated return self.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/defaulttags.py", line 238, in render nodelist.append(node.render_annotated(context)) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 958, in render_annotated return self.render(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 1059, in render output = self.filter_expression.resolve(context) File "/opt/netbox/venv/lib/python3.8/site-packages/django/template/base.py", line 739, in resolve new_obj = func(obj, *arg_vals) File "/opt/netbox/netbox/utilities/templatetags/helpers.py", line 131, in divide return round(x / y) TypeError: unsupported operand type(s) for /: 'str' and 'int'

The problem was originally observed in 3.3.7, and for grins, I upgraded the development environment to 3.3.8. If I create a new PDU with the same device type, I have no problem accessing it.

There were no problems in the upgrade to 3.3.7, or 3.3.4 and 3.3.1 before that.

Where should I look next or grab information from?

**Reply by iunderwood** [reply]
As it turns out, there is something in the render with how the power feed is calculated:

 <td>{{ powerfeed.available_power|divide:3 }}VA</td>

If I look at the connected power feed, there is no available_power field, which implies this must be calculated somehow.  Perhaps something is off in the calculations for 3-phase power, because that seems to be common among the PDUs I can't access.

The power feed object for the issue I reported:

> {
  "id": 230,
  "url": "https://netboxdev/api/dcim/power-feeds/230/",
  "display": "NIT2SKS2B1-25,27,29",
  "power_panel": {
    "id": 28,
    "url": "https://netboxdev/api/dcim/power-panels/28/",
    "display": "NIT2SKS2B1",
    "name": "NIT2SKS2B1"
  },
  "rack": {
    "id": 262,
    "url": "https://netboxdev/api/dcim/racks/262/",
    "display": "MCR-A04",
    "name": "MCR-A04"
  },
  "name": "NIT2SKS2B1-25,27,29",
  "status": {
    "value": "active",
    "label": "Active"
  },
  "type": {
    "value": "redundant",
    "label": "Redundant"
  },
  "supply": {
    "value": "ac",
    "label": "AC"
  },
  "phase": {
    "value": "three-phase",
    "label": "Three-phase"
  },
  "voltage": 208,
  "amperage": 30,
  "max_utilization": 80,
  "comments": "",
  "mark_connected": false,
  "cable": {
    "id": 3393,
    "url": "https://netboxdev/api/dcim/cables/3393/",
    "display": "#3393",
    "label": ""
  },
  "cable_end": "A",
  "link_peers": [
    {
      "id": 1368,
      "url": "https://netboxdev/api/dcim/power-ports/1368/",
      "display": "208v-In",
      "device": {
        "id": 3244,
        "url": "https://netboxdev/api/dcim/devices/3244/",
        "display": "MCRA04RPDU2",
        "name": "MCRA04RPDU2"
      },
      "name": "208v-In",
      "cable": 3393,
      "_occupied": true
    }
  ],
  "link_peers_type": "dcim.powerport",
  "connected_endpoints": [
    {
      "id": 1368,
      "url": "https://netboxdev/api/dcim/power-ports/1368/",
      "display": "208v-In",
      "device": {
        "id": 3244,
        "url": "https://netboxdev/api/dcim/devices/3244/",
        "display": "MCRA04RPDU2",
        "name": "MCRA04RPDU2"
      },
      "name": "208v-In",
      "cable": 3393,
      "_occupied": true
    }
  ],
  "connected_endpoints_type": "dcim.powerport",
  "connected_endpoints_reachable": true,
  "tags": [],
  "custom_fields": {},
  "created": "2021-12-07T00:00:00Z",
  "last_updated": "2022-02-01T20:21:50.118206Z",
  "_occupied": true
}

**Reply by iunderwood** [reply]
Found the problem.

On the RPDUs I have problems with, the power port did not have a value for maximum_draw defined and the form does not allow a zero-value to be defined.  My suspicion is that this is being treated as a string/null and not a zero for the purposes of the calculation, which would explain why the TypeError would show the difference between str and int.

**Comment by iunderwood** [comment]
Secondary note.
When I look at a given power feed, the link for the trace button erroneously links to the interfaces/{id}/trace and not the power-feeds/{id}/trace.

**Comment by kkthxbye-code** [comment]
https://github.com/netbox-community/netbox/pull/10961

Fixed in 3.3.8

Edit: It is fixed but in the upcomming 3.3.9.

**Reply by kkthxbye-code** [reply]
@jeremystretch - seems like the change was added to the release notes after 3.3.8 was released, can you add it to the github release notes?

**Reply by iunderwood** [reply]
This is not fixed in 3.3.8.  Upgrading was the first thing I tried.

**Reply by iunderwood** [reply]
The condition seems to be limited to 3-phase PDUs when there is nothing in the maximum draw field.

**Reply by kkthxbye-code** [reply]
My bad, I misread your post. Feel free to create an issue with the reproduction steps you posted below.

**Comment by iunderwood** [comment]
I'll write this up in the official format tomorrow if that is preferred, but for this thread:

To recreate:
* Create an site, rack, power panel.
* Create a 3-phase power feed.  208, 30A, 80%
* Create a 3-phase PDU device type with one power port, and one power outlet on each phase.
* Create a PDU based upon the previously created device type.
* Connect the power port to the power feed.
* Access the device.

Workaround.
* Add "power-ports/" to the end of the URL.
* Append a non-zero value to "Maximum Draw".
* Access the device.

**Reply by kkthxbye-code** [reply]
Just fyi, I know you haven't made an issue yet, I'm unable to replicate it using the provided steps on 3.3.8.

**Reply by iunderwood** [reply]
I do need to make this more explicit.  I was able to duplicate on the demo site yesterday, so I'll have that report in shortly.

**Reply by iunderwood** [reply]
Error is in with steps to reproduce on the demo site, along with the traceback and debug info from my dev box in this thread.
